### PR TITLE
Make /history works for disruption api

### PR DESCRIPTION
**Pull request by kun2985**
# Description

This PR make /history works for disruption api

## Issue (optional)

Issue link:
bot-1384

### CODE REVIEW

File: chaos/mapper.py [hunk 1]
```diff
@@ -195,3 +195,58 @@ def fill_from_json(item, json, fields):
...
+    disruption.created_at = get_datetime_from_json_attr(json, 'created_at')
+    disruption.updated_at = get_datetime_from_json_attr(json, 'updated_at')
+    disruption.reference = json['reference']
+    disruption.start_publication_date = get_datetime_from_json_attr(json['publication_period'], 'begin')
+    disruption.end_publication_date = get_datetime_from_json_attr(json['publication_period'], 'end')
+    disruption.publicationStatus = json['publication_status']
+    disruption.contributor = contributor
+    disruption.cause = cause
+    disruption.tags = disruption_tags
+    disruption.history_localization = json['localization']
```
Comment (tchevily) [inline chaos/mapper.py]: Why did you add a new attribute to the disruption ? there is already an attribute ''localization''.

File: chaos/fields.py [hunk 2]
```diff
@@ -154,6 +154,11 @@ def output(self, key, obj):
 class FieldLocalization(fields.Raw):
     def output(self, key, obj):
         to_return = []
+
+        #for history
+        if hasattr(obj, 'history_localization') and isinstance(obj.localizations, list):
```
Comment (tchevily) [inline chaos/fields.py]: Why did you add a new attribute to the disruption ? there is already an attribute ''localization''.

File: chaos/history.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+def save_in_database(disruption_id, disruption_json):
+    history_disruption = HistoryDisruption()
+
+    history_disruption.disruption_id = disruption_id
+    history_disruption.data = disruption_json
+    db.session.add(history_disruption)
+    db.session.commit()
+
+def clean_before_save_in_history(disruption):
+    for key in disruption.keys():
```
Comment (Netmisa) [inline chaos/history.py]: To fix error when we have multiple channel

```suggestion
    for key in disruption.keys():
        if isinstance(disruption, str):
        return
```

File: chaos/history.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+def save_disruption_in_history(data):
+    old_display_impacts = g.get('display_impacts')
+    g.display_impacts = True
+    disruption = marshal(data, disruption_fields)
+    g.display_impacts = old_display_impacts
+
+    clean_before_save_in_history(disruption)
+    disruption['impacts'] = disruption['impacts']['impacts']
+    save_in_database(data.id, json.dumps(disruption))
+    print json.dumps(disruption)
```
Comment (Netmisa) [inline chaos/history.py]: Forgotten `print`

```suggestion
```

File: chaos/mapper.py [hunk 5]
```diff
@@ -195,3 +195,58 @@ def fill_from_json(item, json, fields):
...
 }
+
+def get_datetime_from_json_attr(json, attr):
+    return parse_datetime(json[attr]).replace(tzinfo=None) if attr in json and json[attr] else None
+
+
+def disruption_from_history(disruption, json):
+    contributor = models.Contributor()
+    contributor.contributor_code = json['contributor']
+
```
Comment (Netmisa) [inline chaos/mapper.py]:
```suggestion
```

File: chaos/history.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+def save_in_database(disruption_id, disruption_json):
+    history_disruption = HistoryDisruption()
+
+    history_disruption.disruption_id = disruption_id
+    history_disruption.data = disruption_json
+    db.session.add(history_disruption)
+    db.session.commit()
+
+def clean_before_save_in_history(disruption):
+    for key in disruption.keys():
```
Comment (Netmisa) [inline chaos/history.py]:
```suggestion
    if not isinstance(disruption, dict) and not isinstance(disruption, list):
        return
    for key in disruption.keys():
```

File: chaos/history.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+    history_disruption.disruption_id = disruption_id
+    history_disruption.data = disruption_json
+    db.session.add(history_disruption)
+    db.session.commit()
+
+def clean_before_save_in_history(disruption):
+    if not isinstance(disruption, dict) and not isinstance(disruption, list):
+        return
+    if not isinstance(disruption, dict) or not isinstance(disruption, list):
```
Comment (Netmisa) [inline chaos/history.py]:
```suggestion
```

File: chaos/history.py [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+        return
```
Comment (Netmisa) [inline chaos/history.py]:
```suggestion
```

**Comment by kun2985** [comment]
jenkins, retest this please
